app/graph.py

```diff
@@ -470,23 +488,111 @@ def update(self):
-                    'neighbours': {},
-                    'labels': {}
+            try:
+                node_name = i.spec.node_name
+                if node_name:
+                    nodes = self.topology['nodes']
+                    if node_name not in nodes:
+                        nodes[node_name] = {
+                            "node_ip": i.status.host_ip,
+                            "pods": {},
+                            'bgp_peers': {},
+                            'neighbours': {},
+                            'labels': {},
+                            'node_leaf_sec_iface_conn': [],
+                            'node_pod_sec_iface_conn': [],
+                            'node_leaf_ter_iface_conn': [],
+                            'node_pod_ter_iface_conn': [],
+                            'node_leaf_all_iface_conn': [],
+                        }
+
+                    pods = nodes[node_name]['pods']
+                    pod_name = i.metadata.name
+                    pods[pod_name] = {
+                        "ip": i.status.pod_ip,
+                        "primary_iface": "",
+                        "ns": i.metadata.namespace,
+                        "labels": i.metadata.labels if i.metadata.labels is not None else {},
+                        "other_ifaces": {},
+                        "annotations": i.metadata.annotations if i.metadata.annotations is not None else {},
                     }
 
-                self.topology['nodes'][i.spec.node_name]['pods'][i.metadata.name] = {
-                    "ip": i.status.pod_ip,
-                    "ns": i.metadata.namespace,
-                    "labels": i.metadata.labels if i.metadata.labels is not None else {}
-                    }
-                    
+                    annotations = i.metadata.annotations
+                    if annotations is not None:
+                        for key, annotation in annotations.items():
+                            if key == "k8s.v1.cni.cncf.io/network-status":
+                                items_list = json.loads(annotation)
+                                pod_ip = i.status.pod_ip
+                                for val in items_list:
+                                    if val["ips"][0] != pod_ip:
+                                        iface_name = str(val["name"].split('/')[-1])
+                                        pod_iface = str(val["interface"])
+                                        pods[pod_name]['other_ifaces'][iface_name] = pod_iface
+                                    else:
+                                        pods[pod_name]['primary_iface'] = str(val["interface"])
+            except Exception as e:
+                logger.error(f"Error processing pod: {i.metadata.name}. Error: {str(e)}")
+
+        cr = self.custom_obj.list_namespaced_custom_object(group="aci.fabricattachment", version="v1", namespace="aci-containers-system", plural="nodefabricnetworkattachments")
+
+        if cr:
```

Comment:
you need to check if len(cr) == 0 as it is a list

Reply:
I am not able to view any topology even when it is false. Can you check the logic again here